charts/argus/templates/_collector.tpl

```diff
@@ -99,3 +99,31 @@ capabilities:
 {{- end }}
 {{- end }}
 {{- end }}
+
+{{/* This template is used for case checking of COLLECTOR_NON_ROOT env variable */}}
+{{- define "non-root-prop-check"}}
+{{$keyCheckFlag := false}}
+{{$valueCheckFlag := false}}
+{{- $trueCheckList := list "true" "1" "True" "yes" "Yes"}}
+{{- $falseCheckList := list "false" "0" "False" "no" "No"}}
+{{- if not (empty .Values.collector.env)}}
+{{- range $key, $value := .Values.collector.env }}
+    {{- if eq (upper $key) "COLLECTOR_NON_ROOT"}}
+        {{- if eq $key "COLLECTOR_NON_ROOT"}}
+            {{- $keyCheckFlag = false}}
+            {{- if and (not (empty $value)) (not (or (has $value $falseCheckList) (has $value $trueCheckList))) }}
+            {{- $valueCheckFlag = true}}
+            {{- end}}
+        {{- else }}
```

Comment:
I guess this will never be reached ???

Reply:
When the key is not going to be exactly equal to "COLLECTOR_NON_ROOT", this statement will be reached.